alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -253,6 +280,40 @@ public static function file_put_contents ( FileSyncKey $key , $content , $strict
 		self::setPermissions($fullPath);
 		self::createSyncFileForKey($rootPath, $filePath,  $key , $strict , !is_null($res), false, md5($content), kPathManager::getStorageProfileIdForKey($key));
 		self::encryptByFileSyncKey($key, kPathManager::getStorageProfileIdForKey($key));
+		self::setInSmallFileCache($key, $content);
+	}
+
+	protected static function setInSmallFileCache($key, $content)
+	{
+		$fileSyncFilesToCache = kConf::get(self::FILE_SYNC_TYPES_TO_CACHE, kConfMapNames::RUNTIME_CONFIG, array());
+		$maxFileSizeToCache = kConf::get(self::MAX_FILE_SIZE_TO_CACHE, kConfMapNames::RUNTIME_CONFIG);
+		$objectKeys = array (
+			$key->getObjectType() . ':' . $key->getObjectSubType(),
+			$key->getObjectType() . ':' . '*',
+			'*' // WildCard
+		);
+		if (!$fileSyncFilesToCache || !$maxFileSizeToCache || !array_intersect($objectKeys, $fileSyncFilesToCache))
+		{
+			return;
+		}
+
+		$contentSize = strlen($content);
+		if ($contentSize <= $maxFileSizeToCache)
+		{
+			$cacheStore = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_SMALL_FILE_SYNC);
+			if (!$cacheStore)
+			{
+				return;
+			}
+
+			$cacheKey = self::getCacheKey($key);
+			$contentToPut = gzcompress($content) ?? $content;
+			$result = $cacheStore->set($cacheKey, $contentToPut, self::FILE_SYNC_CACHE_EXPIRY);
```

Comment:
add prefix to know if content compressed or not, prefix #COMRESS. or #UNCOMPRES.